src/window/building/depot.c

```diff
@@ -0,0 +1,441 @@
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+    {0, 0, 0, ROW_HEIGHT, button_none, button_none, 0, 0},
+};
+
+static generic_button depot_select_resource_buttons[] = {
+    {18, 0, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN},
+    {232, 0, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 1},
+    {18, 26, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 2},
+    {232, 26, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 3},
+    {18, 52, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 4},
+    {232, 52, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 5},
+    {18, 78, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 6},
+    {232, 78, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 7},
+    {18, 104, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 8},
+    {232, 104, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 9},
+    {18, 130, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 10},
+    {232, 130, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 11},
+    {18, 156, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 12},
+    {232, 156, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 13},
+    {18, 182, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 14},
+    {232, 182, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 15},
+    {18, 208, 214, 26, set_order_resource, button_none, 0, RESOURCE_MIN + 16},
+};
+
+static generic_button depot_order_buttons[] = {
+    {100, 0, 26, 26, order_set_resource, button_none, 1, 0},
+    {100, 56, 284, 22, order_set_source, button_none, 2, 0},
+    {100, 82, 284, 22, order_set_destination, button_none, 3, 0},
+    {100, 30, 284, 22, order_set_condition_type, button_none, 4, 0},
+    {384, 30, 32, 22, order_set_condition_threshold, button_none, 5, 0},
+};
+
+static int storage_buildings_count(void)
+{
+    int count = 0;
+    building *b = building_first_of_type(BUILDING_GRANARY);
+    for (building *b = building_first_of_type(BUILDING_GRANARY); b; b = b->next_of_type) {
+        count++;
+    }
+    for (building *b = building_first_of_type(BUILDING_WAREHOUSE); b; b = b->next_of_type) {
+        count++;
+    }
+    return count;
+}
```

Comment:
This will also count inactive buildings, which isn't what you want.

I'm not sure `building_count_total` does what you need, but you need at least to check against things like `BUILDING_STATE_INACTIVE` or `BUILDING_STATE_DELETED_BY_PLAYER`/`GAME`.

Reply:
Done.